packages/entity-server/src/__tests__/__integration__/EntityRelationshipsRoutes.test.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { TestableEntityServer, setupTestApp, tearDownTestApp } from '../testUtilities';
+import { COUNTRIES } from './fixtures';
+
+describe('relationships', () => {
+  let app: TestableEntityServer;
+
+  beforeAll(async () => {
+    app = await setupTestApp();
+    app.grantAccessToCountries(COUNTRIES);
+  });
+
+  afterAll(() => tearDownTestApp(app));
+
+  describe('/hierarchy/<hierarchyName>/<entityCode>/relationships', () => {
+    describe('errors', () => {
+      it('throws error if no type in descendant_filter', async () => {
+        const { text } = await app.get('hierarchy/redblue/redblue/relationships', {
+          query: { descendant_filter: 'name==Kanto' },
+        });
+
+        const error = JSON.parse(text);
+
+        expect(error).toEqual({
+          error: 'Internal server error: descendant_filter:type url parameter must be present',
```

Comment:
As an optional suggestion, I think it's generally good to not use the exact error messages since they are prone to change. My personal rule of thumb is make them as generic as possible while they still convey their basic message.

```suggestion
          error: 'descendant_filter:type url parameter must be present',
```

Reply:
Good call! Switched!